### PR TITLE
Use less __local memory in P2P

### PR DESCRIPTION
**Pull request by isuruf**

### CODE REVIEW

**Comment by isuruf** [comment]
This needs https://github.com/inducer/loopy/pull/736

**Comment by isuruf** [comment]
GPU tests are running on https://gitlab.tiker.net/inducer/sumpy/-/merge_requests/153

**Comment by isuruf** [comment]
Opened https://github.com/inducer/loopy/issues/770

File: sumpy/p2p.py [hunk 1]
```diff
@@ -473,33 +476,49 @@ def get_kernel(self, max_nsources_in_one_box, max_ntargets_in_one_box,
...
-        if gpu:
+        # This is the amount of targets we process per thread
+        ntgt_offset = (max_ntargets_in_one_box - 1) // work_items_per_group + 1
+        # This is the amount of sources we pre-fetch per thread
+        nsrc_offset_prefetch = (max_nsources_in_one_box - 1) \
+                // work_items_per_group + 1
+        # This is the amount of sources we process per pre-fetch
+        nsrc_offset = (max_nsources_in_one_box - 1) // nprefetch + 1
+        # This is the amount of sources we pre-fetch per thread per pre-fetch
+        nsources_prefetch_per_thread = (nsrc_offset_prefetch - 1) // nprefetch + 1
```
Comment (inducer) [inline sumpy/p2p.py:489]: IMO, this type of thing is exactly what `split_iname` is for. Did that not work?

Comment (isuruf) [reply sumpy/p2p.py:489]: I spent yesterday trying to and the least horrible way to do this with transformations was with https://github.com/isuruf/sumpy/pull/new/less_local2
Which do you prefer?

**Comment by isuruf** [comment]
This is ready now

**Comment by inducer** [comment]
Thanks for working on this!

**Pull request merged**